instance/instance.go

```diff
@@ -398,6 +398,20 @@ func (s *Store) UpdateImportTask(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if tasks.BuildSearchIndexTasks != nil {
+		for _, task := range tasks.BuildSearchIndexTasks {
+			if task.State != "" {
+				if task.State != models.CompletedState {
+					validationErrs = append(validationErrs, fmt.Errorf("bad request - invalid task state value: %v", task.State))
+				} else if err := s.UpdateBuildSearchTaskState(id, task.DimensionName, task.State); err != nil {
+					log.Error(err, nil)
+					http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
We could be returning a mongodb error to the client here. Should we give a more generic error back to the client?